### DISCUSSION TITLE
Description of the programmer stm32flash

### DISCUSSION THREAD

**Discussion by ogogon** [stm32duino/Arduino_Core_STM32#1936] [General]
Please tell me why there is no description for the stm32 flash programmer in the configurations for the Blue Pill?
How to do it right?

Grateful for help,
Ogogon.

**Comment by fpistm** [comment]
All upload methods are described here:
https://github.com/stm32duino/wiki/wiki/Upload-methods

**Reply by ogogon** [reply]
Thank you! This is an interesting document.
However, unfortunately, it does not answer the question I asked, since it does not contain a word about stm32flash.

I noticed, that in your kernel the description of the programmers is not done the way it is usually done in the Arduino IDE.
The programmers.txt file is zero-length, and programmers are described in other ways.

What do I need to do to add a description of my programmer to this structure of yours? I would like to add to the Blue Pill a description of the stm32flash that I am used to, including non-standard options, since I use hardware of my own development, which automatically puts the Pill into programming mode.

**Comment by fpistm** [comment]
Currently, we do not provide any programmer in txt file. You can add it on your own: https://github.com/stm32duino/wiki/wiki/Custom-board-based-on-a-core
